Goal: canonicalize mount point names when umount.cifs is called, to avoid
unnecessarily leaving entries behind in /etc/mtab if invoked with a trailing
slash in the mount point name

Upstream status: submitted as bugzilla bug #4781

Index: samba-3.0.25b/source/client/umount.cifs.c
===================================================================
--- samba-3.0.25b.orig/source/client/umount.cifs.c	2007-07-11 20:33:17.000000000 -0700
+++ samba-3.0.25b/source/client/umount.cifs.c	2007-07-11 20:39:23.000000000 -0700
@@ -342,6 +342,10 @@
 
 	/* fixup path if needed */
 
+	/* Trim any trailing slashes */
+	while (mountpoint[strlen(mountpoint)-1] == '/')
+		mountpoint[strlen(mountpoint)-1] = '\0';
+
 	/* make sure that this is a cifs filesystem */
 	rc = statfs(mountpoint, &statbuf);
 	
